src/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -143,7 +143,7 @@ public RestoreCommand()
             string projectJsonPath = null;
 
             IEnumerable<string> externalProjects = null;
-            if (string.Equals(PackageSpec.PackageSpecFileName, projectFileName, StringComparison.OrdinalIgnoreCase))
+            if (BuildIntegratedProjectUtility.IsProjectConfig(projectPath))
```

Comment:
Would be nice to describe what values `projectPath` can take. Seems it's either xxx.project.json, or something like xxx.csproj